### ISSUE TITLE
Recommend my project

### ISSUE THREAD

**Issue by neozhaoliang** [aosabook/500lines#270]
Hi: I have a [python project](https://github.com/neozhaoliang/gifmaze) that I think may fit in this repo. It builds an environment for running various maze animations from the bottom (in pure python, with only built-in modules). It can produce very nice GIF images. It's roughly 500 lines (the code can be trimmed to be less than 500 lines). Hope you could check it and give suggestions!

**Comment by storyoftime66** [comment]
Page not found now

**Comment by neozhaoliang** [comment]
Yes it's deleted since I thought the maintainers of this project might not be interseted in it and have a look at there, because I got no reply for a year ...

The project [is now here](https://github.com/neozhaoliang/pywonderland/tree/master/src/gifmaze), note it's independent of other directories in that repo, and it has been enlarged since l opened this issue, added more rich features...  but the main framework (in`/gifmaze/`) can be easily fitted into 500 lines.

The code is written in pure python, no extra libs/softwares are needed, and there is no draw/coloring command in the code. It can produce highly optimized gif images in only a few seconds, even there are thousands of frames in the image.

Some examples:

![enterprise](https://user-images.githubusercontent.com/23307174/55521968-9b889680-56b5-11e9-8de6-6d2be02c0ed2.gif)

![gosperglidergun](https://user-images.githubusercontent.com/23307174/55521971-a0e5e100-56b5-11e9-8084-5cdca417d4ff.gif)
![hilbert](https://user-images.githubusercontent.com/23307174/55521974-a6dbc200-56b5-11e9-9bdd-8a05e77ae124.gif)

![langton_ant](https://user-images.githubusercontent.com/23307174/55521980-aba07600-56b5-11e9-8a32-c0a8980be6bd.gif)

![wilson-bfs](https://user-images.githubusercontent.com/23307174/55521993-b529de00-56b5-11e9-9f2f-2508824636a2.gif)
![kruskal](https://user-images.githubusercontent.com/23307174/55521997-b824ce80-56b5-11e9-98fe-e8e591064ef2.gif)

You can run the `example*.py` scripts to see how fast it runs. 

I think this project fits the spirit of 500 lines because:

1. It can output lots of cool gif images, so people can easily enjoy it.
2. It builds everything from the ground (it implemented a simple gif encoder), no third-party libs nor image process softwares are required.
3. Many algorithms are involved here (LZW, Wilson, BFS/DEF, cellualr automata ...)
